### ISSUE TITLE
Provider Road Map for cluster subcomponent resources types

### ISSUE THREAD

**Issue by mkolb-navican** [rancher/terraform-provider-rancher2#246]
Is there a road map to add additional resources types?

There seem to be high level resources for cluster integration, but none of the cluster subcomponents:

Persistent Volumes
Storage Class
Workloads
Load Balancing
Service Discovery
Volumes
Pipelines

**Comment by rawmind0** [comment]
Not by the moment. All subcomponent, except `pipelines`, could be managed by [kubernetes tf provider](https://www.terraform.io/docs/providers/kubernetes/index.html)

**Comment by mkolb-navican** [comment]
That is unfortunate, since subcomponents that are managed with the Kubernetes provider wouldn't be visible within the Rancher UI. This creates a disconnect between the actual system state in code and the application UI.

**Comment by rawmind0** [comment]
@mkolb-navican , k8s subcomponents should be visible within Rancher UI, no matter if they are created from Rancher, kubectl or k8s tf provider.

**Comment by jd4883** [comment]
I started spinning up my own RKE cluster recently and am using terraform to manage it as close to completely as possible. Storage Class / Persistent Volume Storage is one of the few areas I found to not have a good way to manage with terraform. I'm using vsphere as my storage provider and similar to the RKE UI, I would really like to have a terraform block I can define that I can put the vsphere attributes in. The kubernetes provider does not have the same vsphere visibility so while it may be possible to make the configuration work, its harder to pinpoint.

This is one of the few features in my cluster I cannot find a good way to manage in terraform. These additional resource types would be awesome to see support come into a future build!
